src/prefect/tasks/kubernetes/deployment.py

```diff
@@ -87,7 +88,9 @@ def run(
                 "A dictionary representing a ExtensionsV1beta1Deployment must be provided."
             )
 
-        kubernetes_api_key = Secret(kubernetes_api_key_secret).get()
+        kubernetes_api_key = None
+        if kubernetes_api_key_secret:
+            kubernetes_api_key = Secret(kubernetes_api_key_secret).get()
```

Comment:
So the way I read this, by _default_ the key is still required, this just allows you to explicitly provide `None` if you want to bypass it?  I assume needing the key is still the more common use case?

Reply:
Yeah the way I believe this should be used is through the API key and remote connection

Reply:
But you can override it by explicitly providing None

Reply:
Ok cool that makes sense